test/partition.test.js

```diff
@@ -28,141 +29,188 @@ describe('cloudant - partitioned db', () => {
+          done();
         });
-        should.exist(index);
-        index.name.should.equal(indexName);
-        index.def.fields[0]['prodName'].should.equal('asc');
-        index.def.fields[1][DEFAULT_MODEL_VIEW].should.equal('asc');
-        // should be a global index
-        index.partitioned.should.equal(true);
-        done();
       });
     });
-  });
 
-  it('index entry - ' +
-    'create global index for when `partitioned` is configured as false',
-  (done) => {
-    Product = db.define('Product', {
-      prodName: {type: String},
-      desc: {type: String},
-    }, {
-      indexes: {
-        'prodName3_index': {
-          partitioned: false,
-          keys: {
-            prodName: 1,
+    it('index entry - ' +
+      'create global index for when `partitioned` is configured as false',
+    (done) => {
+      Product = db.define('Product', {
+        prodName: {type: String},
+        desc: {type: String},
+      }, {
+        indexes: {
+          'prodName3_index': {
+            partitioned: false,
+            keys: {
+              prodName: 1,
+            },
           },
         },
-      },
+      });
+      db.automigrate('Product', (err) => {
+        if (err) return done(err);
+        connector.getIndexes(connector.getDbName(connector), (e, results) => {
+          if (e) return done(e);
+          const indexes = results.indexes;
+          const indexName = 'prodName3_index';
+          should.exist(indexes);
+
+          const index = _.find(indexes, function(index) {
+            return index.name === indexName;
+          });
+          should.exist(index);
+          index.name.should.equal(indexName);
+          index.def.fields[0]['prodName'].should.equal('asc');
+          index.def.fields[1][DEFAULT_MODEL_VIEW].should.equal('asc');
+          // should be a global index
+          index.partitioned.should.equal(false);
+          done();
+        });
+      });
     });
-    db.automigrate('Product', (err) => {
-      if (err) return done(err);
-      connector.getIndexes(connector.getDbName(connector), (e, results) => {
-        if (e) return done(e);
-        const indexes = results.indexes;
-        const indexName = 'prodName3_index';
-        should.exist(indexes);
+  });
 
-        const index = _.find(indexes, function(index) {
-          return index.name === indexName;
+  context('findAll tests', ()=> {
+    it('find all records by partition key from option', (done) => {
+      Product = db.define('Product', {
+        name: {type: String},
+        tag: {type: String},
+      }, {
+        forceId: false,
+        indexes: {
+          'product_name_index': {
+            partitioned: true,
+            keys: {
+              city: 1,
+            },
+          },
+        },
+      });
+      db.automigrate('Product', () => {
+        Product.create(SEED_DATA, function(err) {
+          if (err) return done(err);
+          Product.find({where: {tag: 'food'}}, {partitionKey: 'toronto'},
```

Comment:
Just out of curious, I guess the performance of partition + where(#L196) is better than doing where on find() normally?